x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2817,3 +2860,9 @@ function parseDate(dateString: string | undefined, propertyName: string, default
 
   return parsedDate;
 }
+
+function clearUnscheduledSnooze(attributes: { snoozeSchedule?: RuleSnooze }) {
```

Comment:
Not completely clear what this function is doing, but I think it's removing "mute all" snoozes that have a duration.

Reply:
Not exactly. This clears any "simple" snoozes, i.e. the user selects "Snooze for 3 days (starting now)" in the currently-existing UX, instead of e.g. "Snooze on Friday for 3 days" or "Snooze every Friday for 3 days" as made possible by the new functionality. It's the functional equivalent of setting `snoozeEndTime: null` in the previous version of the data model.